bindings/rust/src/lib.rs

```diff
@@ -139,6 +139,38 @@ impl From<ExecutionResult> for sys::FizzyExecutionResult {
 }
 
 impl Instance {
+    fn checked_memory_range(&self, offset: u32) -> core::ops::Range<usize> {
+        let offset = offset as usize;
+        let size = 0 as usize;
+        offset..offset + size
+    }
+
+    pub fn memory_size(&self) -> usize {
+        unsafe { sys::fizzy_get_instance_memory_size(self.0.as_ptr()) }
+    }
+
+    pub fn memory_get_into(&mut self, offset: u32, target: &mut [u8]) -> Result<(), ()> {
+        let mem = unsafe {
```

Comment:
Return err if no memory found (`fizzy_get_instance_memory_data` returns null).

Reply:
More errors at the beginning sound good.